sdk/armcore/connection_test.go

```diff
@@ -128,3 +131,40 @@ func TestScope(t *testing.T) {
 		t.Fatalf("unexpected scope %s", s)
 	}
 }
+
+func TestDisableAutoRPRegistration(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	// initial response that RP is unregistered
+	srv.SetResponse(mock.WithStatusCode(http.StatusConflict), mock.WithBody([]byte(rpUnregisteredResp)))
+	con := NewConnection(srv.URL(), mockTokenCred{}, &ConnectionOptions{DisableRPRegistration: true})
+	if ep := con.Endpoint(); ep != srv.URL() {
+		t.Fatalf("unexpected endpoint %s", ep)
+	}
+	req, err := azcore.NewRequest(context.Background(), http.MethodGet, srv.URL())
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	resp, err := con.Pipeline().Do(req)
```

Comment:
Why is this called twice? (also on line 159)

Reply:
Copy/paste error, fixed.